src/Symfony/Component/Yaml/Inline.php

```diff
@@ -398,7 +398,7 @@ private static function parseSequence($sequence, $flags, &$i = 0, $references =
                     $value = self::parseScalar($sequence, $flags, array(',', ']'), array('"', "'"), $i, true, $references);
 
                     // the value can be an array if a reference has been resolved to an array var
-                    if (!is_array($value) && !$isQuoted && false !== strpos($value, ': ')) {
+                    if (!is_array($value) && !$value instanceof \DateTimeInterface && !$isQuoted && false !== strpos($value, ': ')) {
```

Comment:
What about using `!is_object()` here instead? That would also cover the case when an unserialized object was returned before.